crates/ruff_server/src/session/settings.rs

```diff
@@ -306,11 +356,15 @@ impl ResolvedClientSettings {
             ),
             editor_settings: ResolvedEditorSettings {
                 configuration: Self::resolve_optional(all_settings, |settings| {
-                    settings
-                        .configuration
-                        .as_ref()
-                        .and_then(|config_path| shellexpand::full(config_path).ok())
-                        .map(|config_path| PathBuf::from(config_path.as_ref()))
+                    settings.configuration.as_ref().and_then(|configuration| {
+                        match ResolvedConfiguration::try_from(configuration) {
+                            Ok(configuration) => Some(configuration),
+                            Err(err) => {
+                                tracing::error!("Failed to resolve configuration: {err}");
+                                None
+                            }
+                        }
+                    })
```

Comment:
I'm planning to open a follow-up PR to notify the user that resolving the client settings failed. We currently don't log or notify the user if it fails. For example, `lint.select = ["RUF07", "I001"]` will see that `RUF07` fails and thus ignores this config value which means that `I001` is not selected either.